scheduler/node/node.go

```diff
@@ -81,3 +89,32 @@ func (n *Node) AddContainer(container *cluster.Container) error {
 	n.Containers = append(n.Containers, container)
 	return nil
 }
+
+// ReserveResource reserve some resources for a container
+func (n *Node) ReserveResource(config *cluster.ContainerConfig) error {
+	n.Lock()
+	defer n.Unlock()
+	if config != nil {
+		memory := config.Memory
+		cpus := config.CpuShares
+		if (n.TotalMemory-n.UsedMemory)-memory < 0 || (n.TotalCpus-n.UsedCpus)-cpus < 0 {
+			return ErrNotEnoughResources
+		}
+		n.UsedMemory = n.UsedMemory + memory
+		n.UsedCpus = n.UsedCpus + cpus
+	}
+	return nil
+}
+
+// ReleaseResource releases resource on the node
+func (n *Node) ReleaseResource(config *cluster.ContainerConfig) error {
+	n.Lock()
+	defer n.Unlock()
+	if config != nil {
+		memory := config.Memory
+		cpus := config.CpuShares
+		n.UsedMemory = n.UsedMemory - memory
```

Comment:
I think you should check that this is >= 0, otherwise something really wrong happened